moshi-kotlin-codegen/src/main/java/com/squareup/moshi/kotlin/codegen/api/ProguardRules.kt

```diff
@@ -70,14 +70,11 @@ public data class ProguardConfig(
     if (targetConstructorHasDefaults) {
       // If the target class has default parameter values, keep its synthetic constructor
       //
-      // -keepnames class kotlin.jvm.internal.DefaultConstructorMarker
       // -keepclassmembers @com.squareup.moshi.JsonClass @kotlin.Metadata class * {
       //     synthetic <init>(...);
       // }
       //
       appendLine("-if class $targetName")
-      appendLine("-keepnames class kotlin.jvm.internal.DefaultConstructorMarker")
```

Comment:
We should keep this, we still want to allow R8 to minify when it's truly never used

Reply:
`DefaultConstructorMarker` class is just a flag, keeping the name means keeping everything.  I want to obfuscate it with R8.

Reply:
I have removed all relevant keep rules, which means it is auto deleted when not in use.